src/docker-in-docker/install.sh

```diff
@@ -365,10 +440,18 @@ install_previous_version_artifacts() {
     if [ $wget_exit_code -eq 8 ]; then  # failure due to 404: Not Found.
         echo -e "\n(!) Failed to fetch the latest artifacts for docker buildx v${buildx_version}..."
         repo_url="https://api.github.com/repos/docker/buildx/releases" # GitHub repository URL
-        previous_version=$(get_previous_version "${repo_url}")
-        buildx_file_name="buildx-${previous_version}.linux-${architecture}"
-        echo -e "\nAttempting to install ${previous_version}"
-        wget https://github.com/docker/buildx/releases/download/${previous_version}/${buildx_file_name}
+        err_msg=""
+        get_previous_version "${repo_url}" buildx_version err_msg
+        if [[ "${err_msg}" == *"Rate limit exceeded. Fallback implemented."* ]]; then
```

Comment:
In future, let's avoid this kind of code logic where a conditional block is dependent on previously called function's error message. 

Messages are dynamic and can be accidentally updated which will end up breaking this

Reply:
sure, will keep this in mind for future code changes ! thanks !